cli/azd/pkg/spin/run.go

```diff
@@ -2,13 +2,31 @@ package spin
 
 import (
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/theckman/yacspin"
 )
 
+// SpinnerUpdater contains callbacks for updating the spinner.
+type SpinnerUpdater struct {
```

Comment:
wondering if we should make something different about the spinner...

We started by passing a callback fn() to be wrapped between spinner.start and stop.   
Now we are adding callbacks for stop, log and start.
I see a complicated way of callbacks for us if we continue like that.

I wonder if we should create an entire stateful component to wrap the spinner and provide an easy API to do what we want...  Just wondering....  Let me know what you think...

Reply:
I was thinking the same, but didn't do a refactor since I was including in the same PR. It's still currently maintainable, but definitely won't bode well if we continue the same pattern.

I'll break this change up. The `SpinnerUpdater` here is already essentially the component that wraps the spinner (in fact I should just rename it to reflect it). Then if we separate out the creation vs running, we'd achieve the goal of separation of concerns.

Reply:
Submitted #166